test/y2storage/dump_manager_test.rb

```diff
@@ -0,0 +1,251 @@
+    end
+
+    describe "#devicegraph_dump_name" do
+      it "Can handle a nil devicegraph" do
+        expect(subject.devicegraph_dump_name(nil)).to be_nil
+      end
+
+      it "Returns 'probed' for the probed devicegraph" do
+        expect(subject.devicegraph_dump_name(probed)).to eq "probed"
+      end
+
+      it "Returns 'staging' for the staging devicegraph" do
+        expect(subject.devicegraph_dump_name(probed)).to eq "probed"
+      end
+
+      it "Returns 'devicegraph' for anything else" do
+        expect(subject.devicegraph_dump_name(Object.new)).to eq "devicegraph"
+      end
+    end
+
+    describe "#dump" do
+      it "Can handle a nil object" do
+        expect(subject.dump(nil)).to be_nil
+      end
+
+      it "Correctly dumps the probed devicegraph" do
+        expect(dir?("storage")).to be false
+        expect(subject.dump(probed)).to eq "01-probed"
+        expect(dump?("01-probed")).to be true
+      end
+
+      it "Correctly dumps the staging devicegraph" do
+        expect(subject.dump(staging)).to eq "02-staging"
+        expect(dump?("02-staging")).to be true
+        expect(dump?("01-probed")).to be true
+      end
+
+      it "Correctly dumps the staging devicegraph again" do
+        expect(subject.dump(staging)).to eq "03-staging"
+        expect(dump?("03-staging")).to be true
+        expect(dump?("02-staging")).to be true
+        expect(dump?("01-probed")).to be true
+      end
+
+      it "Correctly dumps the staging devicegraph again to a specified name" do
+        expect(subject.dump(staging, "committed")).to eq "04-committed"
+        expect(dump?("04-committed")).to be true
+        expect(dump?("03-staging")).to be true
+        expect(dump?("02-staging")).to be true
+        expect(dump?("01-probed")).to be true
+      end
+
+      it "Correctly dumps an ActionsPresenter" do
+        expect(subject.dump(Y2Storage::ActionsPresenter.new(nil))).to eq "05-actions"
+        expect(File.exist?(base_dir + "/storage/05-actions.txt")).to be true
+      end
+
+      it "Rejects unknown types" do
+        expect { subject.dump(Object.new) }.to raise_error(ArgumentError)
+      end
+    end
+  end
+
+  context "During installation" do
+    let(:dump_dir) { base_dir + "/storage-inst" }
+
+    before(:each) do
+      allow(Yast::Mode).to receive(:installation).and_return(true)
+    end
+
+    describe "#reset" do
+      # It's a singleton - we need to reset it now for Mode.installation to
+      # take effect
+      it "clears storage-inst/" do
+        subject.reset
+        expect(dir?("storage-inst")).to be true
+        expect(marker?("storage-inst", "marker-inst")).to be false
+      end
+
+      it "leaves storage/, storage-03/, ... alone" do
+        expect(marker?("storage", "01-probed.yml")).to be true
+        expect(marker?("storage", "03-staging.yml")).to be true
+        expect(marker?("storage", "04-committed.yml")).to be true
+        expect(marker?("storage-03", "marker-01")).to be true
```

Comment:
I don't understand this test. Why all these files exist?

Reply:
Because of `populate_full` earlier.

Reply:
For those log rotating things it should be obvious; the comments as well as the output of the individual tests already say it.